design/proposals/spike-cli-dockerless.md

```diff
@@ -0,0 +1,257 @@
+
+#### Deployment of Ruby CLI
+
+To deploy the Ruby CLI, a certain Ruby version needs to be installed on the machine. Because we cannot assume machines have connection to the internet, we will need to either figure out how to either:
+
+1. Create an executable that contains all dependencies.
+2. Create a package that will install the CLI and all its dependencies. For REHL and Centos, we would supply users with a RPM package. For macOS, we would supply them with PKG and so on. We will need to provide different installation packages for each platform (rpm, pkg, deb, etc).
+3. Create zip that will extract all content into a target folder and load CLI and its dependencies.
+
+Ruby bumps versions often and it is likely that the newer version of platforms will come with the newer Ruby versions. The newer version of Ruby might break a dependency we relay on to run the CLI. If this is the case, the Ruby CLI will not work, making it difficult to make this Ruby CLI portable. We can request that users install a certain version but changing Ruby versions is not a non-trival undertaking.
+
+#### Deployment of Go CLI
+
+The Go CLI is compiled down to a single executable binary. Go gives us the [ability to cross-build executables](https://www.digitalocean.com/community/tutorials/how-to-build-go-executables-for-multiple-platforms-on-ubuntu-16-04) so we don't need to hop to specific platform to compile the code. 
+
+#### Deployment of Python CLI
+
+The Python CLI is [single executable](https://github.com/cyberark/conjur-api-python3/releases/tag/v0.0.5), packed with the required runtime and dependencies. For this option there is no need for Python or other prerequisites to be installed on machine. Each executable will need to be bundled on each supported platform. 
+
+***NOTE***: Many platforms (except for Linux) require signed executables so might be an extra step in the deployment process
+
+|            | Testing                                                      |
+| ---------- | ------------------------------------------------------------ |
+| Ruby CLI   | Tested very well                                             |
+| Go CLI     | Pipeline maintained but the testing is not at the level it needs to be. Existing functionality will demand additional testing. <br />Considering the limited implemented functionality, new tests will need to be added if we choose to expand. |
+| Python CLI | Tested very well                                             |
+
+None of our CLIs test against other platforms besides Linux.
+
+|            | Security |
+| ---------- | -------- |
+| Ruby CLI   | See scan |
+| Go CLI     | See scan |
+| Python CLI | See scan |
+
+#### CLI commands to be supported
+
+The CLI should support the following commands:
+
+*Core* -  `init`, `authn` (`login`, `authenticate`),  `policy (load)`, `list`, `show `, `rotate_api_key` (`host` / `user`) and `fetch secrets`
```

Comment:
Again, maybe focus on user stories or REST APIs instead of concrete commands?

Reply:
@alexkalish the requirements are regarding the UX of the CLI not REST API.
The list will be re examine with PM, PS and so on.. but it is important to list the capabilities the CLI provides.

Reply:
@InbalZilberman: I understand that this is for the CLI.  However, this says: "The CLI should support the following commands:".  Why do they have to be named like that?  Why is `init` the right choice?  I think it's better to speak in terms of functionality than concrete commands, otherwise you risk biasing reads towards existing commands/UXs.

Reply:
@alexkalish the idea of the epic is NOT to redo UX it is to have the core functionality provided today to our customers provided also deployed in a machine without docker

Reply:
does this mean not redoing the Python UX, if we move forward with that CLI as a Dockerless option? because its functionality for configuring the CLI is _leagues_ ahead of what the Ruby CLI uses.

Reply:
@izgeri I copied over your input to the new PR where I have updated the doc based on feedback mentioned here. Feel free to take a look! See [here](https://github.com/cyberark/conjur/pull/1917/files#diff-94fda9f3ac9c67857646893fe6d3fd0053a6b319cd511dc03744f648e3da293dR45)
I agree we should start thinking about our CLI in terms of business logic and not make a replica of the Ruby CLI if we realize already that it is bad UX. It from the start limits us and narrows our thinking